python/pyspark/sql/pandas/serializers.py

```diff
@@ -183,6 +212,21 @@ def create_array(s, t):
                     raise e
             return array
 
+        def to_plain_struct(cell):
```

Comment:
The performance here will be very bad. We should create a function based on the type, and avoid type-dispatching for every value.

Reply:
let me see what i can do.